### PR TITLE
feat(exec, execd): add an option to pass a custom environment to their child process

### LINKED ISSUE #9281: `inputs.exec`, `inputs.execd`: add an option to pass a custom environment to their child process
### Proposal:
Add an option in `inputs.exec` and `inputs.execd` to pass custom environment variables to their child process.

### Current behavior:
Currently, if you want to pass some values to the child process, you can only use the command line parameters or the whole environment of Telegraf.

### Desired behavior:
For `inputs.execd`, there is always a single command so we could have an option like this:
```
[[inputs.execd]]

command = ["/path/to/my/process"]

[inputs.execd.env]
CUSTOM_VAR_1 = "var1"
CUSTOM_VAR_2 = "var2"
CUSTOM_VAR_2 = "var3"
```

### Use case: 
This can be useful to easily customize the child process behavior without having a separate configuration file and not display this information in the command line (e.g. for credentials)

### PR DESCRIPTION
**Pull request by an-mmx**
resolves #9281

adds support for environment variables for **execd** and **exec** plugins

example usage:
```toml
[[inputs.execd]]
  command = [
    "/opt/custom-plugins/my-custom-plugin",
    "-config", "/opt/custom-plugins/my-custom-plugin.conf",
    "-interval", "30s",
  ]
  environment = ["LD_LIBRARY_PATH=/opt/app/lib64"]
```

### CODE REVIEW

File: plugins/inputs/exec/README.md [hunk 1]
```diff
@@ -17,6 +17,10 @@ This plugin can be used to poll for custom metrics from any source.
     "/tmp/collect_*.sh"
   ]
 
+  ## Environment variables
+  ## eg: environment = ["LD_LIBRARY_PATH=/opt/custom/lib64:/usr/local/libs", "USERNAME=John Doe"]
+  environment = ["LD_LIBRARY_PATH=/opt/custom/lib64"]
```
Comment (powersj) [inline plugins/inputs/exec/README.md:22]: The value in the sample config should be the default and commented out. In this case, we don't want everyone's config exporting that library path.

```suggestion
  ## Array of "key=value" pairs to pass as environment variables
  ## e.g. "KEY=value", "USERNAME=John Doe",
  ## "LD_LIBRARY_PATH=/opt/custom/lib64:/usr/local/libs"
  # environment = []
```

File: plugins/inputs/execd/README.md [hunk 2]
```diff
@@ -22,6 +22,10 @@ STDERR from the process will be relayed to Telegraf as errors in the logs.
   ## NOTE: process and each argument should each be their own string
   command = ["telegraf-smartctl", "-d", "/dev/sda"]
 
+  ## Environment variables
+  ## eg: environment = ["LD_LIBRARY_PATH=/opt/custom/lib64:/usr/local/libs", "USERNAME=John Doe"]
+  environment = ["LD_LIBRARY_PATH=/opt/custom/lib64"]
```
Comment (powersj) [inline plugins/inputs/execd/README.md:27]:
```suggestion
  ## Array of "key=value" pairs to pass as environment variables
  ## e.g. "KEY=value", "USERNAME=John Doe",
  ## "LD_LIBRARY_PATH=/opt/custom/lib64:/usr/local/libs"
  # environment = []
```

File: plugins/outputs/exec/README.md [hunk 3]
```diff
@@ -20,6 +20,10 @@ For better performance, consider execd, which runs continuously.
   ## Command to ingest metrics via stdin.
   command = ["tee", "-a", "/dev/null"]
 
+  ## Environment variables
+  ## eg: environment = ["LD_LIBRARY_PATH=/opt/custom/lib64:/usr/local/libs", "USERNAME=John Doe"]
+  # environment = ["LD_LIBRARY_PATH=/opt/custom/lib64"]
```
Comment (powersj) [inline plugins/outputs/exec/README.md:25]:
```suggestion
  ## Array of "key=value" pairs to pass as environment variables
  ## e.g. "KEY=value", "USERNAME=John Doe",
  ## "LD_LIBRARY_PATH=/opt/custom/lib64:/usr/local/libs"
  # environment = []
```

File: plugins/outputs/execd/README.md [hunk 4]
```diff
@@ -13,6 +13,10 @@ Telegraf minimum version: Telegraf 1.15.0
   ## NOTE: process and each argument should each be their own string
   command = ["my-telegraf-output", "--some-flag", "value"]
 
+  ## Environment variables
+  ## eg: environment = ["LD_LIBRARY_PATH=/opt/custom/lib64:/usr/local/libs", "USERNAME=John Doe"]
+  environment = ["LD_LIBRARY_PATH=/opt/custom/lib64"]
```
Comment (powersj) [inline plugins/outputs/execd/README.md:18]:
```suggestion
  ## Array of "key=value" pairs to pass as environment variables
  ## e.g. "KEY=value", "USERNAME=John Doe",
  ## "LD_LIBRARY_PATH=/opt/custom/lib64:/usr/local/libs"
  # environment = []
```

**Review by powersj — CHANGES_REQUESTED** [review]
Hi!

This is pretty sweet work! Thank you for putting this all together and with tests.

My only concern is with the sample config in each of the README. I have suggested an update, but feel free to let me know if you think otherwise.

Thanks again!

**Review by powersj — APPROVED** [review]

**Review by reimda — APPROVED** [review]

**Pull request merged**